pageserver/src/object_repository.rs

```diff
@@ -632,7 +632,7 @@ impl Timeline for ObjectTimeline {
         }))
     }
 
-    fn gc_iteration(&self, horizon: u64) -> Result<GcResult> {
+    fn gc_iteration(&self, horizon: u64, compact: bool) -> Result<GcResult> {
```

Comment:
Please, add some comments explaining when we should or shouldn't pass the compact flag. Is it only for tests?

Other than that PR looks good to me.

Reply:
Done, but comment is added to trait method definition in repository.rs